SelfAndPeerEvaluations/partner_eval_quiz_validator.py

```diff
@@ -0,0 +1,457 @@
+                "assignment_group_id": self.assignment.assignment_group_id,
+                "submission_types": ["none"],
+                "points_possible": 1,
+                "due_at": datetime.datetime.now().isoformat(),
+                "lock_at": datetime.datetime.now().isoformat(),
+                "unlock_at": datetime.datetime.now().isoformat(),
+                "published": True,
+                "omit_from_final_grade": True,
+            }
+        )
+        grade_data = defaultdict(dict)
+        for student_id in self.quiz_grades["info"]["submissions"]:
+            if student_id in self.quiz_errors:
+                grade_data[student_id]["posted_grade"] = 0
+                grade_data[student_id]["text_comment"] = ",\n".join(
+                    [str(error) for error in self.quiz_errors[student_id]]
+                )
+            else:
+                grade_data[student_id]["posted_grade"] = 1
+        validation_assignment.submissions_bulk_update(grade_data=grade_data)
+        if len(self.quiz_errors.keys()) == 0:  # all students submitted perfectly
+            return
+        else:
+            self.assignment.create_override(
+                assignment_override={
+                    "student_ids": list(self.quiz_errors.keys()),
+                    "title": f"{self.assignment.name} make up",
+                    "unlock_at": datetime.datetime.now(),
+                    "due_at": datetime.datetime.now() + datetime.timedelta(days=2),
+                    "lock_at": datetime.datetime.now() + datetime.timedelta(days=2),
+                }
+            )
+
+    def __log_solo_submissions(self) -> None:
+        """
+        log solo submssion id's and justifications to a .txt file
+        """
+        for index, json_question in enumerate(self.json_questions):
+            if json_question["question_name"] == "Solo Submission Justification":
+                question_id = self.canvas_questions[index]["id"]
+        justification_index = None
+
+        outfile = open(
+            f"./quiz_results/solo_submissions/{self.assignment.name}_solo_submissions.txt",
+            "w",
+        )
+        outfile.writelines("id (name): justification\n")
+        for user_id in self.solo_submission_ids:
+            submission = self.assignment.get_submission(
+                user_id, include="submission_history"
+            )
+            submission_data = submission.submission_history[0]["submission_data"]
+            if justification_index is None:
+                for index, _ in enumerate(submission_data):
+                    if _["question_id"] == int(question_id):
+                        justification_index = index
+
+            justification = submission_data[justification_index]["text"]
```

Comment:
Should probably strip the justification of whitespace so something like "   " also counts as No Justification Provided